grpc.go

```diff
@@ -322,6 +322,60 @@ func (a *agentGRPC) updateContainerConfig(spec *specs.Spec, config *configs.Conf
 	return a.updateContainerConfigPrivileges(spec, config)
 }
 
+func updateGRPCSpecDevices(mounts []*pb.Storage, spec *pb.Spec) error {
+	if spec.Linux == nil || len(spec.Linux.Devices) == 0 {
+		// No reason to go further in this case since we won't be
+		// able to modify any device from the spec.
+		return nil
+	}
+
+	for _, mnt := range mounts {
+		if mnt == nil {
+			continue
+		}
+
+		if mnt.Driver == "" {
+			continue
+		}
```

Comment:
In this case, what's the content of mnt.Driver?
I think this code needs to be moved to the corresponding storage driver.
The behavior of this function will be a part of the ABI between runtime <-> agent.
So it needs to be documented in protocol or storage driver.

Reply:
Well I wanted to raise this PR to discuss if you were okay using this field in order to determine if we actually have a mount or a device. I am assuming an empty driver will mean we have a mount, vs a device if the driver field contains something. Let me know if you're okay with this (@laijs @bergwolf @jodh-intel @sameo), and I will update the doc in the protocol.
Another option would be to define a new field/structure `Device` that we would pass the same way we pass `Storage` for the mounts.

Reply:
1) register drivers implementations to the agent as a list
2) if the driver field contains something, search the driver from the list and call the handling function

Reply:
> I am assuming an empty driver will mean we have a mount, vs a device if the driver field contains something

If we agree on this, please can you add that comment to the definition to make this assumption clear to all:

- https://github.com/kata-containers/agent/blob/master/protocols/grpc/agent.proto#L197